src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -347,22 +206,22 @@ public void createModificationGroup(UUID sourceGroupUuid, UUID groupUuid) {
         }
     }
 
-    // This function cannot be @Transactional because we clone all modifications resetting their id to null,
-    // which is not allowed by JPA if we still stay in the same Tx.
-    public List<UUID> duplicateModifications(UUID targetGroupUuid, UUID networkUuid, UUID reportUuid, UUID reporterId, String variantId, List<UUID> modificationsUuidList) {
-        List<ModificationEntity> duplicatedModificationEntityList = new ArrayList<>();
-        List<UUID> missingModificationList = new ArrayList<>();
-        for (UUID modifyId : modificationsUuidList) {
-            networkModificationRepository.cloneModificationEntity(modifyId).ifPresentOrElse(
-                duplicatedModificationEntityList::add,
-                () -> missingModificationList.add(modifyId)  // data no more available
-            );
-        }
-        if (!duplicatedModificationEntityList.isEmpty()) {
-            networkModificationRepository.saveModifications(targetGroupUuid, duplicatedModificationEntityList);
+    @Transactional
+    public List<UUID> duplicateModifications(UUID targetGroupUuid, NetworkInfos networkInfos, ReportInfos reportInfos, List<UUID> modificationsUuids) {
+        List<ModificationEntity> modificationsEntities = networkModificationRepository.getModificationsEntities(modificationsUuids);
+        Set<UUID> presentUuids = modificationsEntities.stream().map(ModificationEntity::getId).collect(Collectors.toSet());
+        List<ModificationEntity> duplicatedModificationsEntities = modificationsEntities.stream().map(ModificationEntity::copy).collect(Collectors.toList());
+        List<UUID> missingModificationList = new ArrayList<>(modificationsUuids);
+        missingModificationList.removeAll(presentUuids);
+        if (!duplicatedModificationsEntities.isEmpty()) {
+            networkModificationRepository.saveModifications(targetGroupUuid, duplicatedModificationsEntities);
             // try to apply the duplicated modifications (incremental mode)
-            List<UUID> duplicatedModificationList = duplicatedModificationEntityList.stream().map(ModificationEntity::getId).collect(Collectors.toList());
-            applyModifications(duplicatedModificationList, targetGroupUuid, networkUuid, reportUuid, reporterId, variantId);
+            if (networkInfos.isVariantPresent()) { // TODO return NetworkDamages() ?
```

Comment:
TODO return NetworkDamages()
is it still necessary ? I thing the "damages" is already managed by the applicator class

Reply:
Same  above